lib/src/live_chat_platform_interface.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:live_chat/src/live_chat_method_channel.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+abstract class LiveChatPlatform extends PlatformInterface {
+  LiveChatPlatform() : super(token: _token);
+
+  static final Object _token = Object();
+
+  static LiveChatPlatform _instance = MethodChannelLiveChat();
+
+  /// The default instance of [LiveChatPlatform] to use.
+  ///
+  /// Defaults to [MethodChannelLiveChat].
+  static LiveChatPlatform get instance => _instance;
+
+  /// Platform-specific implementations should set this with their own
+  /// platform-specific class that extends [LiveChatPlatform] when
+  /// they register themselves.
+  static set instance(LiveChatPlatform instance) {
+    PlatformInterface.verifyToken(instance, _token);
+    _instance = instance;
+  }
+
+  Future<void> openChatView({
+    required String licenseId,
+    required String username,
+    required String email,
+    String? groupId,
+    Map<String, String>? customParameters,
+  }) {
+    throw UnimplementedError('openChatView() has not been implemented.');
+  }
+
+  Future<void> closeChatView() {
+    throw UnimplementedError('closeChatView() has not been implemented.');
+  }
+
+  Future<void> clearChatView() {
+    throw UnimplementedError('clearChatView() has not been implemented.');
+  }
+
+  Stream<dynamic>? getEventsStream() {
+    throw UnimplementedError('getEventsStream() has not been implemented.');
+  }
```

Comment:
This class is abstract, means it's not gonna be instantiated.
Can we keep the methods abstracts as well, without implementations. this will also enforce each Platform specific class to implement the methods.